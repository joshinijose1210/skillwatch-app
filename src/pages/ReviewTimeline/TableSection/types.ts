export interface TableSectionProps {
    data: any;
}
