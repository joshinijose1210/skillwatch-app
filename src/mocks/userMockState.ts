export const userMockState = {
    id: 72,
    organisationId: 1,
    firstName: 'ABC',
    lastName: 'DEF',
    email: 'abc@scalereal.com',
    phoneNumber: '+917060753645',
    employeeId: 'SR0022',
    roleName: 'Manager',
    teamName: 'BA',
    modulePermission: [
        {
            moduleId: 16,
            moduleName: 'Request Feedback',
            view: true,
            edit: true
        },
        {
            moduleId: 3,
            moduleName: 'Feedback',
            view: true,
            edit: true
        },
        {
            moduleId: 10,
            moduleName: 'Performance Review',
            view: true,
            edit: true
        },
        {
            moduleId: 18,
            moduleName: 'Analytics',
            view: true,
            edit: true
        },
        {
            moduleId: 20,
            moduleName: 'Departments',
            view: true,
            edit: true
        },
        {
            moduleId: 4,
            moduleName: 'Teams',
            view: true,
            edit: true
        },
        {
            moduleId: 5,
            moduleName: 'Designations',
            view: true,
            edit: true
        },
        {
            moduleId: 6,
            moduleName: 'Roles & Permissions',
            view: true,
            edit: true
        },
        {
            moduleId: 7,
            moduleName: 'Employees',
            view: true,
            edit: true
        },
        {
            moduleId: 8,
            moduleName: 'KPIs',
            view: true,
            edit: true
        },
        {
            moduleId: 12,
            moduleName: 'Reviews for Team Members',
            view: true,
            edit: true
        },
        {
            moduleId: 13,
            moduleName: 'Check-in with Team Members',
            view: true,
            edit: true
        },
        {
            moduleId: 9,
            moduleName: 'Review Cycles',
            view: true,
            edit: true
        },
        {
            moduleId: 15,
            moduleName: 'Company Information',
            view: true,
            edit: false
        },
        {
            moduleId: 17,
            moduleName: 'Allowed Domains',
            view: true,
            edit: true
        },
        {
            moduleId: 19,
            moduleName: 'Integrations',
            view: true,
            edit: true
        },
        {
            moduleId: 14,
            moduleName: 'User Activity Log',
            view: true,
            edit: true
        },
        {
            moduleId: 19,
            moduleName: 'KRAs',
            view: true,
            edit: true
        },
        {
            moduleId: 18,
            moduleName: 'Received Suggestions',
            view: true,
            edit: true
        }
    ],
    designationName: 'BA - Traineee',
    isLoggedIn: true,
    onboardingFlow: true
};

export const empUserState = {
    ...userMockState,
    roleName: 'Employee',
    modulePermission: [
        {
            moduleId: 16,
            moduleName: 'Request Feedback',
            view: true,
            edit: true
        },
        {
            moduleId: 3,
            moduleName: 'Feedback',
            view: true,
            edit: true
        },
        {
            moduleId: 10,
            moduleName: 'Performance Review',
            view: false,
            edit: false
        },
        {
            moduleId: 18,
            moduleName: 'Analytics',
            view: false,
            edit: false
        },
        {
            moduleId: 20,
            moduleName: 'Departments',
            view: false,
            edit: false
        },
        {
            moduleId: 4,
            moduleName: 'Teams',
            view: false,
            edit: false
        },
        {
            moduleId: 5,
            moduleName: 'Designations',
            view: false,
            edit: false
        },
        {
            moduleId: 6,
            moduleName: 'Roles & Permissions',
            view: false,
            edit: false
        },
        {
            moduleId: 7,
            moduleName: 'Employees',
            view: false,
            edit: false
        },
        {
            moduleId: 8,
            moduleName: 'KPIs',
            view: false,
            edit: false
        },
        {
            moduleId: 12,
            moduleName: 'Reviews for Team Members',
            view: false,
            edit: false
        },
        {
            moduleId: 13,
            moduleName: 'Check-in with Team Members',
            view: false,
            edit: false
        },
        {
            moduleId: 9,
            moduleName: 'Review Cycles',
            view: false,
            edit: false
        },
        {
            moduleId: 15,
            moduleName: 'Company Information',
            view: false,
            edit: false
        },
        {
            moduleId: 17,
            moduleName: 'Allowed Domains',
            view: false,
            edit: false
        },
        {
            moduleId: 19,
            moduleName: 'Integrations',
            view: false,
            edit: false
        },
        {
            moduleId: 14,
            moduleName: 'User Activity Log',
            view: false,
            edit: false
        }
    ]
};

export const orgAdminMockState = {
    id: 1,
    organisationId: 1,
    firstName: 'ABC',
    lastName: 'DEF',
    email: 'abc@scalereal.com',
    phoneNumber: '+917060753645',
    employeeId: 'SR0002',
    roleName: 'Org Admin',
    teamName: 'Org Admin',
    modulePermission: [
        {
            moduleId: 6,
            moduleName: 'Feedback',
            view: true,
            edit: false
        },
        {
            moduleId: 7,
            moduleName: 'Teams',
            view: true,
            edit: true
        },
        {
            moduleId: 8,
            moduleName: 'Designations',
            view: true,
            edit: true
        },
        {
            moduleId: 9,
            moduleName: 'Roles & Permissions',
            view: true,
            edit: true
        },
        {
            moduleId: 10,
            moduleName: 'Employees',
            view: true,
            edit: true
        },
        {
            moduleId: 11,
            moduleName: 'KPIs',
            view: true,
            edit: true
        },
        {
            moduleId: 14,
            moduleName: 'Review Cycles',
            view: true,
            edit: true
        },
        {
            moduleId: 17,
            moduleName: 'Performance Review',
            view: true,
            edit: false
        },
        {
            moduleId: 12,
            moduleName: 'Reviews for Team Members',
            view: true,
            edit: true
        },
        {
            moduleId: 13,
            moduleName: 'Check-in with Team Members',
            view: true,
            edit: true
        },
        {
            moduleId: 16,
            moduleName: 'User Activity Log',
            view: true,
            edit: false
        },
        {
            moduleId: 15,
            moduleName: 'Company Information',
            view: true,
            edit: true
        },
        {
            moduleId: 1,
            moduleName: 'Request Feedback',
            view: true,
            edit: true
        },
        {
            moduleId: 2,
            moduleName: 'Settings',
            view: true,
            edit: true
        },
        {
            moduleId: 4,
            moduleName: 'Integrations',
            view: true,
            edit: true
        },
        {
            moduleId: 3,
            moduleName: 'Analytics',
            view: true,
            edit: true
        },
        {
            moduleId: 5,
            moduleName: 'Departments',
            view: true,
            edit: true
        },
        {
            moduleId: 18,
            moduleName: 'Received Suggestions',
            view: true,
            edit: true
        },
        {
            moduleId: 19,
            moduleName: 'KRAs',
            view: true,
            edit: true
        }
    ],
    designationName: 'Org Admin',
    isLoggedIn: true,
    onboardingFlow: true
};
