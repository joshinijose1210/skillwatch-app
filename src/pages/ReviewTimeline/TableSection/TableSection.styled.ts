import styled from 'styled-components';

export const RatingWrapper = styled.div`
    display: flex;
    align-items: center;
    gap: 5px;
`;

export const RatingTable = styled.div`
    width: 380px;
`;

export const RatingTitle = styled.div`
    width: 70%;
`;

export const RatingChipContainer = styled.div`
    width: 70%;
`;
