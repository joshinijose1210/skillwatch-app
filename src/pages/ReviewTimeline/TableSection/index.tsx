export { TableSection as default } from './TableSection';
