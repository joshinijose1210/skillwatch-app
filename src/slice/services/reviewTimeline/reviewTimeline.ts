import apiUrls from '@constants/apiUrls';
import { createApi } from '@reduxjs/toolkit/query/react';
import { axiosBaseQuery } from '@utils/axiosBaseQuery';
import { createUrlWithParams } from '@utils/createUrlWithParams';

export const reviewTimelineApi = createApi({
    reducerPath: 'reviewTimelineApi',
    baseQuery: axiosBaseQuery({
        baseUrl: apiUrls.reviewTimeline
    }),
    keepUnusedDataFor: 0.0001,
    tagTypes: ['timeline'],

    endpoints: builder => ({
        getReviewTimeline: builder.query({
            query: payload => ({
                url: createUrlWithParams(payload)
            }),
            providesTags: ['timeline']
        }),
        getTimeline: builder.mutation({
            query: payload => ({
                method: 'GET',
                url: createUrlWithParams(payload)
            })
        })
    })
});

export const { useGetReviewTimelineQuery, useGetTimelineMutation } = reviewTimelineApi;
