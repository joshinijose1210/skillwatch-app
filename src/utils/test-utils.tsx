import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
