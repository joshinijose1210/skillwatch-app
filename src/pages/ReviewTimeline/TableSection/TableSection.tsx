import { RatingChip } from '@components';
import { Grid } from '@components/layout/Grid/Grid';
import { Text } from '@medly-components/core';
import { useAppSelector } from '@slice';
import { FC } from 'react';
import { StyledAccordion, AccordionContent, AccordionHeader, TableSectionWrapper } from '../ReviewTimeline.styled';
import { RatingTable } from './TableSection.styled';
import { TableSectionProps } from './types';

export const TableSection: FC<TableSectionProps> = ({ data }) => {
    const selfAverageRating = (data[0] && data[0].selfAverageRating) || -1,
        firstManagerAverageRating = (data[0] && data[0].firstManagerAverageRating) || -1,
        secondManagerAverageRating = (data[0] && data[0].secondManagerAverageRating) || -1,
        checkInWithManagerAverageRating = (data[0] && data[0].checkInWithManagerAverageRating) || -1;
    const userDetails = useAppSelector(state => state.user);

    return (
        <TableSectionWrapper>
            {!(data && userDetails.roleName === 'Org Admin' && !data[0]?.empDetails) && (
                <StyledAccordion defaultActive>
                    <AccordionHeader>
                        <Text textVariant="h4">Average Rating</Text>
                    </AccordionHeader>
                    <AccordionContent noSidePadding>
                        <RatingTable>
                            <Grid row={true} alignItems="center" marginBottom={1}>
                                <Grid column={true} md={5}>
                                    <Text textVariant="body2" textWeight="Medium">
                                        Self Review:
                                    </Text>
                                </Grid>
                                <Grid column={true}>
                                    <RatingChip rating={selfAverageRating} isInReviewTimeline />
                                </Grid>
                            </Grid>
                            <Grid row={true} alignItems="center" marginBottom={1}>
                                <Grid column={true} md={5}>
                                    <Text textVariant="body2" textWeight="Medium">
                                        Manager 1 Review:
                                    </Text>
                                </Grid>
                                <Grid column={true}>
                                    <RatingChip rating={firstManagerAverageRating} isInReviewTimeline />
                                </Grid>
                            </Grid>
                            <Grid row={true} alignItems="center" marginBottom={1}>
                                <Grid column={true} md={5}>
                                    <Text textVariant="body2" textWeight="Medium">
                                        Manager 2 Review:
                                    </Text>
                                </Grid>
                                <Grid column={true}>
                                    <RatingChip rating={secondManagerAverageRating} isInReviewTimeline />
                                </Grid>
                            </Grid>
                            <Grid row={true} alignItems="center" marginBottom={1}>
                                <Grid column={true} md={5}>
                                    <Text textVariant="body2" textWeight="Medium">
                                        Check-in with Manager Review:
                                    </Text>
                                </Grid>
                                <Grid column={true}>
                                    <RatingChip rating={checkInWithManagerAverageRating} isInReviewTimeline />
                                </Grid>
                            </Grid>
                        </RatingTable>
                    </AccordionContent>
                </StyledAccordion>
            )}
        </TableSectionWrapper>
    );
};
