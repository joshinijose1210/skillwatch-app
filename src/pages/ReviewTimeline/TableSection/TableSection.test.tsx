import { RTKRender, screen } from '@utils/test-utils';
import { TableSection } from './TableSection';
import { reviewTimelineMockData } from '@mocks/reviewTimeline';
import { empUserState, orgAdminMockState } from '@mocks/userMockState';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockNavigate
}));

describe('Table section', () => {
    it('should render as expected', async () => {
        const { data } = reviewTimelineMockData.reviewTimeline.defaultGet;
        const { container } = RTKRender(<TableSection data={data} />);

        expect(container).toMatchSnapshot();
    });

    it('should not show Rating section if the user is Org Admin without members for review', async () => {
        const { data } = reviewTimelineMockData.reviewTimeline.orgAdminWithoutEmpGet;
        RTKRender(<TableSection data={data} />, { initialState: orgAdminMockState });

        expect(screen.queryByText('Average Rating')).not.toBeInTheDocument();
    });

    it('should show self, my manager and my check-in quick links if the user is employee', () => {
        const { data } = reviewTimelineMockData.reviewTimeline.managerReviewPendingGet;
        RTKRender(<TableSection data={data} />, { initialState: empUserState });
        expect(screen.getByText(/4.5/i)).toBeInTheDocument();
    });

    it('should return -1 when ratings are not provided', () => {
        const data = [{}];
        RTKRender(<TableSection data={data} />, { initialState: empUserState });
        expect(screen.getAllByText(/Not yet given./i).length).toBe(4);
    });
});
