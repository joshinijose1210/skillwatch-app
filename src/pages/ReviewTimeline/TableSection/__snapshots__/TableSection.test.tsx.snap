// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table section should render as expected 1`] = `
<div>
  <div
    class="sc-1fchxsp-6 dqYAvH"
  >
    <div
      class="sc-w3sqmf-0 hnFIgR sc-1fchxsp-15"
    >
      <button
        aria-expanded="true"
        class="sc-1ouxg1r-0 hwLnLP sc-1fchxsp-12 HZcpt"
      >
        <h4
          class="sc-1ntgl3l-0 iEGCWU"
          textalign="initial"
          textcolor="inherit"
          textvariant="h4"
        >
          Average Rating
        </h4>
        <svg
          class="sc-1pf3x8a-0 klDMsE"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.875 9l-3.88 3.88L8.115 9a.996.996 0 1 0-1.41 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59a.996.996 0 0 0 0-1.41c-.39-.38-1.03-.39-1.42 0z"
            fill="#000"
            fill-opacity=".54"
          />
        </svg>
      </button>
      <div
        class="sc-aqez41-0 ljhHPp sc-1fchxsp-13 fWXnRP"
        role="region"
      >
        <div
          class="sc-72u0du-1 cnjwts"
        >
          <div
            class="sc-h70dbj-0 fQmYvl row align-center marginBottom-1"
          >
            <div
              class="sc-h70dbj-0 fQmYvl column md-5"
            >
              <span
                class="sc-1ntgl3l-0 gPMgWC"
                textalign="initial"
                textcolor="inherit"
                textvariant="body2"
                textweight="Medium"
              >
                Self Review:
              </span>
            </div>
            <div
              class="sc-h70dbj-0 fQmYvl column"
            >
              <div
                class="sc-1yzztbi-0 sc-1yzztbi-1 hmfzBC jdubJU"
              >
                <button
                  class="sc-1qiq5dc-0 cEdgdr sc-1yzztbi-2 gcoBEt"
                  variant="solid"
                >
                  <span
                    class="sc-1ntgl3l-0 hGagSU"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    0
                  </span>
                </button>
                <span
                  class="sc-1ntgl3l-0 gPMgWC sc-1yzztbi-3 iLVinv"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="body2"
                  textweight="Medium"
                >
                  Not yet given.
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-h70dbj-0 fQmYvl row align-center marginBottom-1"
          >
            <div
              class="sc-h70dbj-0 fQmYvl column md-5"
            >
              <span
                class="sc-1ntgl3l-0 gPMgWC"
                textalign="initial"
                textcolor="inherit"
                textvariant="body2"
                textweight="Medium"
              >
                Manager 1 Review:
              </span>
            </div>
            <div
              class="sc-h70dbj-0 fQmYvl column"
            >
              <div
                class="sc-1yzztbi-0 sc-1yzztbi-1 hmfzBC jdubJU"
              >
                <button
                  class="sc-1qiq5dc-0 cEdgdr sc-1yzztbi-2 gcoBEt"
                  variant="solid"
                >
                  <span
                    class="sc-1ntgl3l-0 hGagSU"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    0
                  </span>
                </button>
                <span
                  class="sc-1ntgl3l-0 gPMgWC sc-1yzztbi-3 iLVinv"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="body2"
                  textweight="Medium"
                >
                  Not yet given.
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-h70dbj-0 fQmYvl row align-center marginBottom-1"
          >
            <div
              class="sc-h70dbj-0 fQmYvl column md-5"
            >
              <span
                class="sc-1ntgl3l-0 gPMgWC"
                textalign="initial"
                textcolor="inherit"
                textvariant="body2"
                textweight="Medium"
              >
                Manager 2 Review:
              </span>
            </div>
            <div
              class="sc-h70dbj-0 fQmYvl column"
            >
              <div
                class="sc-1yzztbi-0 sc-1yzztbi-1 hmfzBC jdubJU"
              >
                <button
                  class="sc-1qiq5dc-0 cEdgdr sc-1yzztbi-2 gcoBEt"
                  variant="solid"
                >
                  <span
                    class="sc-1ntgl3l-0 hGagSU"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    0
                  </span>
                </button>
                <span
                  class="sc-1ntgl3l-0 gPMgWC sc-1yzztbi-3 iLVinv"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="body2"
                  textweight="Medium"
                >
                  Not yet given.
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-h70dbj-0 fQmYvl row align-center marginBottom-1"
          >
            <div
              class="sc-h70dbj-0 fQmYvl column md-5"
            >
              <span
                class="sc-1ntgl3l-0 gPMgWC"
                textalign="initial"
                textcolor="inherit"
                textvariant="body2"
                textweight="Medium"
              >
                Check-in with Manager Review:
              </span>
            </div>
            <div
              class="sc-h70dbj-0 fQmYvl column"
            >
              <div
                class="sc-1yzztbi-0 sc-1yzztbi-1 hmfzBC jdubJU"
              >
                <button
                  class="sc-1qiq5dc-0 cEdgdr sc-1yzztbi-2 gcoBEt"
                  variant="solid"
                >
                  <span
                    class="sc-1ntgl3l-0 hGagSU"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    0
                  </span>
                </button>
                <span
                  class="sc-1ntgl3l-0 gPMgWC sc-1yzztbi-3 iLVinv"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="body2"
                  textweight="Medium"
                >
                  Not yet given.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
